backend/vouch/trust_algo.py

```diff
@@ -1,110 +1,128 @@
+                normalized_vouch_matrix[voucher][vouchee] \
+                    += IMPLICIT_PRETRUST_VOUCH / normalization_constant
+            if vouch_matrix[voucher][vouchee] > 0:
+                normalized_vouch_matrix[voucher][vouchee] += 1 / normalization_constant
+    return normalized_vouch_matrix
 
 
-def get_trust_vector(C, p):
+def compute_relative_posttrusts(normalized_vouch_matrix, relative_pretrusts: NDArray):
     """
     Return a vector of global trust values per user, given the vouchers in the
     network and the set of pre-trusted users. This part comes directly from
     EigenTrust.
     """
-    t = p
-    newt = t
+    relative_trusts = relative_pretrusts
+    new_relative_trusts = relative_trusts
     delta = 10
-    while delta >= EPS:
-        newt = C.T.dot(t)
-        newt = (1 - A) * newt + A * p
-        delta = np.linalg.norm(newt - t)
-        t = newt
-    return newt
+    while delta >= APPROXIMATION_ERROR:
+        new_relative_trusts = normalized_vouch_matrix.T.dot(relative_trusts)
+        new_relative_trusts = (1 - PRETRUST_BIAS) * new_relative_trusts \
+            + PRETRUST_BIAS * new_relative_trusts
+        delta = np.linalg.norm(new_relative_trusts - relative_trusts)
+        relative_trusts = new_relative_trusts
+    return new_relative_trusts
 
 
-def rescale(trust_vector, trust_status):
+def compute_voting_rights(relative_posttrusts, pretrusts):
     """
     Go from ratio of trust to actual voting weight that should be assigned to
     users given the trust the network puts in them.
     """
-    mint = np.amin(
-        [trust_vector[i] for i in range(len(trust_vector)) if trust_status[i] > 0]
+    min_relative_trusts_of_pretrusteds = np.amin(
+        [relative_posttrusts[u] for u in range(len(relative_posttrusts)) if pretrusts[u] > 0]
     )
-    return np.array(trust_vector) / mint
+    scale = MIN_PRETRUST_VOTING_RIGHT / min_relative_trusts_of_pretrusteds
+    scaled_relative_trusts = np.array(relative_posttrusts) * scale
+    clipped_relative_trusts = np.array([
+        min(scaled_relative_trusts[u], 1) for u in range(len(relative_posttrusts))
+    ])
+    return clipped_relative_trusts
 
 
 def trust_algo():
     """
     Improved version of the EigenTrust algorithm.
 
-    Compute a global trust score for each user, based on the set of
-    pre-trusted users* and on vouching made between users.
+    Compute a global trust score for all users, based on the set of
+    pre-trusted users and on vouching made between users.
 
     (* the ones with an email from a trusted domain).
     """
-    # List of users, list of their trust status (regarding their email) in the
-    # same order, and number of users.
+    # Import users and pretrust status
     users = list(
         user.User.objects.all().annotate(
             _is_trusted=Q(pk__in=user.User.trusted_users())
         )
     )
-    trust_status = [int(u._is_trusted) for u in users]
+    pretrusts = [int(u._is_trusted) for u in users]
```

Comment:
To apply the previous suggestions, you may need to initialize these values as a numpy array
```suggestion
    pretrusts = np.array([int(u._is_trusted) for u in users])
```